src/prefect/configuration.py

```diff
@@ -57,19 +58,10 @@ def string_to_type(val: str) -> Union[bool, int, float, str]:
     elif val.upper() == "FALSE":
         return False
 
-    # int
+    # dicts
```

Comment:
```suggestion
    # dicts, ints, floats, or any other literal Python syntax
```

the docstring is also out of date, we can probably rip out the whole "Maps" section since it should work with any literal